Una vez tenemos los datos, vamos acrear tablas, ordenarlos y prepararlos para el análisis

In [1]:
import pandas as pd
import numpy as np

### Creamos data frame de airbnb y ordenamos datos

In [2]:
a_marzo = pd.read_csv("airbnb_marzo_2023", index_col= 0)
a_abril = pd.read_csv("airbnb_abril_2023", index_col= 0)
a_mayo = pd.read_csv("airbnb_mayo_2023", index_col= 0)
a_junio = pd.read_csv("airbnb_junio_2023", index_col= 0)
a_julio = pd.read_csv("airbnb_julio_2023", index_col= 0)
a_agosto = pd.read_csv("airbnb_agosto_2023", index_col= 0)
a_sept = pd.read_csv("airbnb_sept_2023", index_col= 0)
a_oct = pd.read_csv("airbnb_oct_2023", index_col= 0)
a_nov = pd.read_csv("airbnb_nov_2023", index_col= 0)
a_dic = pd.read_csv("airbnb_dic_2023", index_col= 0)
a_enero = pd.read_csv("airbnb_enero_2024", index_col= 0)
a_feb = pd.read_csv("airbnb_feb_2024", index_col= 0)

Como la forma de ordenar los datos que he utilizado para los hoteles no funcionaba con estos, ya que salían los numeros raros y con muchos decimales usé estos pasos para colocar los dataframes. 

In [108]:
#a_feb = list(a_feb.stack())
#a_feb.sort()
#a_feb = pd.DataFrame(a_feb, columns= ["Febrero 2024"])
#a_feb.to_csv("airbnb_feb_2024")

Febrero 2024
0            95.0
1            98.0
2           111.0
3           112.0
4           112.0
..            ...
265         321.0
266         328.0
267         332.0
268         339.0
269         361.0

[270 rows x 1 columns]

In [3]:
datos_airbnb = pd.concat([a_marzo, a_abril, a_mayo, a_junio, a_julio, a_agosto, a_sept, a_oct, a_nov, a_dic, a_enero, a_feb], axis=1)
#datos_airbnb = datos_airbnb.astype(int)
datos_airbnb = datos_airbnb.fillna(method="ffill")
datos_airbnb

Marzo 2023  Abril 2023  Mayo 2023  Junio 2023  Julio 2023  Agosto 2023  \
0         125.0        81.0       81.0       101.0       101.0        129.0   
1         133.0       122.0      133.0       135.0       130.0        130.0   
2         135.0       133.0      142.0       142.0       141.0        135.0   
3         167.0       164.0      143.0       144.0       142.0        135.0   
4         181.0       166.0      152.0       155.0       150.0        135.0   
..          ...         ...        ...         ...         ...          ...   
265       759.0       841.0      430.0       410.0       444.0        328.0   
266       790.0       852.0      452.0       417.0       445.0        331.0   
267       839.0       852.0      466.0       457.0       451.0        337.0   
268       913.0       852.0      544.0       458.0       456.0        376.0   
269       913.0       852.0      621.0       468.0       485.0        416.0   

     Septiembre 2023  Octubre 2023  Noviembre 2023  Diciembre 2023  \
0              111.0         111.0           111.0           111.0   
1              129.0         112.0           112.0           112.0   
2              135.0         117.0           112.0           112.0   
3              139.0         128.0           117.0           117.0   
4              152.0         129.0           128.0           128.0   
..               ...           ...             ...             ...   
265            341.0         357.0           361.0           384.0   
266            357.0         363.0           384.0           398.0   
267            369.0         369.0           405.0           419.0   
268            405.0         384.0           410.0           421.0   
269            458.0         405.0           419.0           432.0   

     Enero 2024  Febrero 2024  
0         111.0          95.0  
1         112.0          98.0  
2         112.0         111.0  
3         112.0         112.0  
4         115.0         112.0  
..          ...           ...  
265       351.0         321.0  
266       361.0         328.0  
267       398.0         332.0  
268       398.0         339.0  
269       419.0         361.0  

[270 rows x 12 columns]

Para ver cual es el rango principal entre los que se encuentran los precios los separamos por rangos

In [10]:
rango_50_150 = []
rango_151_250 = []
rango_251_350 = []
rango_351_450 = []
rango_451_600 = []
rango_resto = []


for i in datos_airbnb.columns:
    for dato in range(len(datos_airbnb[i])):
    
        if datos_airbnb[i][dato] <= 150:
            rango_50_150.append(datos_airbnb[i][dato])
        elif datos_airbnb[i][dato] > 150 and datos_airbnb[i][dato] <= 250:
            rango_151_250.append(datos_airbnb[i][dato])
        elif datos_airbnb[i][dato] > 250 and datos_airbnb[i][dato] <= 350:
            rango_251_350.append(datos_airbnb[i][dato])
        elif datos_airbnb[i][dato] > 350 and datos_airbnb[i][dato] <= 450:
            rango_351_450.append(datos_airbnb[i][dato])
        elif datos_airbnb[i][dato] > 450 and datos_airbnb[i][dato] <= 600:
            rango_451_600.append(datos_airbnb[i][dato])
        else:
            rango_resto.append(datos_airbnb[i][dato])      
    

In [14]:
rango1 = pd.DataFrame(rango_50_150, columns= ["50 - 150"])
rango2 = pd.DataFrame(rango_151_250, columns= ["151 - 250"])
rango3 = pd.DataFrame(rango_251_350, columns= ["251 - 350"])
rango4 = pd.DataFrame(rango_351_450, columns= ["351 - 450"])
rango5 = pd.DataFrame(rango_451_600, columns= ["451 - 600"])
rango6 = pd.DataFrame(rango_resto, columns= ["resto"])

rangos_airbnb = pd.concat([rango1, rango2, rango3, rango4, rango5, rango6], axis=1)
rangos_airbnb

50 - 150  151 - 250  251 - 350  351 - 450  451 - 600  resto
0        125.0      167.0      252.0      351.0      455.0  601.0
1        133.0      181.0      252.0      351.0      455.0  603.0
2        135.0      187.0      253.0      351.0      457.0  603.0
3         81.0      190.0      258.0      351.0      458.0  611.0
4        122.0      193.0      258.0      354.0      458.0  612.0
...        ...        ...        ...        ...        ...    ...
1703       NaN      246.0        NaN        NaN        NaN    NaN
1704       NaN      249.0        NaN        NaN        NaN    NaN
1705       NaN      249.0        NaN        NaN        NaN    NaN
1706       NaN      250.0        NaN        NaN        NaN    NaN
1707       NaN      250.0        NaN        NaN        NaN    NaN

[1708 rows x 6 columns]

### Creamos data frame de hoteles y ordenamos datos

In [26]:
h_marzo = pd.read_csv("hotel_marzo_2023", index_col= 0)
h_abril = pd.read_csv("hotel_abril_2023", index_col= 0)
h_mayo = pd.read_csv("hotel_mayo_2023", index_col= 0)
h_junio = pd.read_csv("hotel_junio_2023", index_col= 0)
h_julio = pd.read_csv("hotel_julio_2023", index_col= 0)
h_agosto = pd.read_csv("hotel_agosto_2023", index_col= 0)
h_sept = pd.read_csv("hotel_sept_2023", index_col= 0)
h_oct = pd.read_csv("hotel_oct_2023", index_col= 0)
h_nov = pd.read_csv("hotel_nov_2023", index_col= 0)
h_dic = pd.read_csv("hotel_dic_2023", index_col= 0)
h_enero = pd.read_csv("hotel_enero_2024", index_col= 0)
h_feb = pd.read_csv("hotel_feb_2024", index_col= 0)

In [27]:
datos_hoteles = pd.concat([h_marzo, h_abril, h_mayo, h_junio, h_julio, h_agosto, h_sept, h_oct, h_nov, h_dic, h_enero, h_feb], axis=1)

Como de airbnb tenemos 270 datos y de hoteles tenemos muchos más, para hacer una comparación dentro de unos precios medios que podría pagar un público de clase media filtramos para que nos eneñe los precios que estén por debajo de 1000 y se pueda hacer un abuena estimación de precios y ver cual puede salir más rentable

In [28]:
datos_hoteles = datos_hoteles[datos_hoteles < 1000]
datos_hoteles = datos_hoteles.fillna(method= "ffill")
datos_hoteles

Marzo 2023  Abril 2023  Mayo 2023  Junio 2023  Julio 2023  Agosto 2023  \
0          259.0       225.0      258.0       228.0       169.0        178.0   
1          259.0       225.0      258.0       228.0       169.0        178.0   
2          259.0       225.0      258.0       228.0       169.0        178.0   
3          330.0       531.0      202.0       528.0       458.0        335.0   
4          310.0       492.0      238.0       528.0       455.0        321.0   
...          ...         ...        ...         ...         ...          ...   
2037       141.0       159.0      334.0       354.0       545.0        467.0   
2038       141.0       171.0      334.0       354.0       545.0        467.0   
2039       141.0       342.0      334.0       354.0       545.0        467.0   
2040       141.0       342.0      334.0       354.0       545.0        467.0   
2041       141.0       184.0      334.0       354.0       545.0        467.0   

      Septiembre 2023  Octubre 2023  Noviembre 2023  Diciembre 2023  \
0               179.0         498.0           478.0           458.0   
1               179.0         470.0           452.0           440.0   
2               179.0         498.0           479.0           459.0   
3               488.0         283.0           238.0           517.0   
4               484.0         283.0           238.0           494.0   
...               ...           ...             ...             ...   
2037            221.0         335.0           420.0           249.0   
2038            221.0         335.0           420.0           249.0   
2039            221.0         335.0           420.0           249.0   
2040            221.0         335.0           420.0           249.0   
2041            221.0         335.0           420.0           249.0   

      Enero 2024  Febrero 2024  
0          438.0         438.0  
1          414.0         414.0  
2          438.0         442.0  
3          622.0         383.0  
4          622.0         383.0  
...          ...           ...  
2037       225.0          74.0  
2038       225.0          74.0  
2039       225.0          74.0  
2040       225.0          74.0  
2041       225.0          74.0  

[2042 rows x 12 columns]

Como no nos interesa en este caso que hotel elegir si no solo los precios y hemos rellenado los valores NaN con precios que ya estan en la lista y se van a repetir, filtramos por los valores unicos de los datos obtenidos

In [29]:
h_marzo1 = pd.DataFrame(datos_hoteles["Marzo 2023"].sort_values().unique(), columns= ["Marzo 2023"])
h_abril1 = pd.DataFrame(datos_hoteles["Abril 2023"].sort_values().unique(), columns= ["Abril 2023"])
h_mayo1 = pd.DataFrame(datos_hoteles["Mayo 2023"].sort_values().unique(), columns= ["Mayo 2023"])
h_junio1 = pd.DataFrame(datos_hoteles["Junio 2023"].sort_values().unique(), columns= ["Junio 2023"])
h_julio1 = pd.DataFrame(datos_hoteles["Julio 2023"].sort_values().unique(), columns= ["Julio 2023"])
h_agosto1 = pd.DataFrame(datos_hoteles["Agosto 2023"].sort_values().unique(), columns= ["Agosto 2023"])
h_sept1 = pd.DataFrame(datos_hoteles["Septiembre 2023"].sort_values().unique(), columns= ["Septiembre 2023"])
h_oct1 = pd.DataFrame(datos_hoteles["Octubre 2023"].sort_values().unique(), columns= ["Octubre 2023"])
h_nov1 = pd.DataFrame(datos_hoteles["Noviembre 2023"].sort_values().unique(), columns= ["Noviembre 2023"])
h_dic1 = pd.DataFrame(datos_hoteles["Diciembre 2023"].sort_values().unique(), columns= ["Diciembre 2023"])
h_enero1 = pd.DataFrame(datos_hoteles["Enero 2024"].sort_values().unique(), columns= ["Enero 2024"])
h_feb1 = pd.DataFrame(datos_hoteles["Febrero 2024"].sort_values().unique(), columns= ["Febrero 2024"])

Al final obtenemos un Data Frame con muchos NaN ya que dependiendo de la fecha no hay l amisma disponibilidad de precios y por lo tanto algunos mese tienen más que otros, pero como lo vamos a analizar por separado no pasa nada. Si no al quitar los NaN se pierden muchos datos de otros meses que pueden interesar

In [30]:
datos_hoteles1 = pd.concat([h_marzo1, h_abril1, h_mayo1, h_junio1, h_julio1, h_agosto1, h_sept1, h_oct1, h_nov1, h_dic1, h_enero1, h_feb1], axis=1)
datos_hoteles1

Marzo 2023  Abril 2023  Mayo 2023  Junio 2023  Julio 2023  Agosto 2023  \
0          90.0        91.0      108.0       108.0       108.0        110.0   
1          92.0        92.0      115.0       115.0       111.0        114.0   
2          98.0        93.0      126.0       126.0       115.0        118.0   
3          99.0        96.0      131.0       138.0       126.0        119.0   
4         102.0        97.0      132.0       145.0       136.0        122.0   
..          ...         ...        ...         ...         ...          ...   
604         NaN       986.0        NaN         NaN         NaN          NaN   
605         NaN       990.0        NaN         NaN         NaN          NaN   
606         NaN       991.0        NaN         NaN         NaN          NaN   
607         NaN       993.0        NaN         NaN         NaN          NaN   
608         NaN       994.0        NaN         NaN         NaN          NaN   

     Septiembre 2023  Octubre 2023  Noviembre 2023  Diciembre 2023  \
0              114.0         108.0           101.0            88.0   
1              122.0         118.0           108.0           109.0   
2              126.0         128.0           112.0           113.0   
3              136.0         139.0           116.0           115.0   
4              138.0         140.0           123.0           117.0   
..               ...           ...             ...             ...   
604              NaN           NaN             NaN             NaN   
605              NaN           NaN             NaN             NaN   
606              NaN           NaN             NaN             NaN   
607              NaN           NaN             NaN             NaN   
608              NaN           NaN             NaN             NaN   

     Enero 2024  Febrero 2024  
0          67.0          52.0  
1          68.0          54.0  
2          76.0          67.0  
3          81.0          68.0  
4          83.0          70.0  
..          ...           ...  
604         NaN           NaN  
605         NaN           NaN  
606         NaN           NaN  
607         NaN           NaN  
608         NaN           NaN  

[609 rows x 12 columns]

In [31]:
#Separamos por rango de precios para ver cual es el que predomina

rango1_50_150 = []
rango1_151_250 = []
rango1_251_350 = []
rango1_351_450 = []
rango1_451_600 = []
rango1_resto = []


for i in datos_hoteles1.columns:
    for dato in range(len(datos_hoteles1[i])):
    
        if datos_hoteles1[i][dato] <= 150:
            rango1_50_150.append(datos_hoteles1[i][dato])
        elif datos_hoteles1[i][dato] > 150 and datos_hoteles1[i][dato] <= 250:
            rango1_151_250.append(datos_hoteles1[i][dato])
        elif datos_hoteles1[i][dato] > 250 and datos_hoteles1[i][dato] <= 350:
            rango1_251_350.append(datos_hoteles1[i][dato])
        elif datos_hoteles1[i][dato] > 350 and datos_hoteles1[i][dato] <= 450:
            rango1_351_450.append(datos_hoteles1[i][dato])
        elif datos_hoteles1[i][dato] > 450 and datos_hoteles1[i][dato] <= 600:
            rango1_451_600.append(datos_hoteles1[i][dato])
        else:
            rango1_resto.append(datos_hoteles1[i][dato])   

In [34]:
rango11 = pd.DataFrame(rango1_50_150, columns= ["50 - 150"])
rango21 = pd.DataFrame(rango1_151_250, columns= ["151 - 250"])
rango31 = pd.DataFrame(rango1_251_350, columns= ["251 - 350"])
rango41 = pd.DataFrame(rango1_351_450, columns= ["351 - 450"])
rango51 = pd.DataFrame(rango1_451_600, columns= ["451 - 600"])
rango61 = pd.DataFrame(rango1_resto, columns= ["resto"])

rangos_hoteles = pd.concat([rango11, rango21, rango31, rango41, rango51, rango61], axis=1)
rangos_hoteles

50 - 150  151 - 250  251 - 350  351 - 450  451 - 600  resto
0         90.0      151.0      252.0      351.0      451.0  602.0
1         92.0      152.0      253.0      352.0      452.0  606.0
2         98.0      153.0      254.0      353.0      454.0  608.0
3         99.0      154.0      256.0      354.0      455.0  609.0
4        102.0      155.0      257.0      355.0      456.0  610.0
...        ...        ...        ...        ...        ...    ...
2661       NaN        NaN        NaN        NaN        NaN    NaN
2662       NaN        NaN        NaN        NaN        NaN    NaN
2663       NaN        NaN        NaN        NaN        NaN    NaN
2664       NaN        NaN        NaN        NaN        NaN    NaN
2665       NaN        NaN        NaN        NaN        NaN    NaN

[2666 rows x 6 columns]